crates/ethereum/evm/src/execute.rs

```diff
@@ -378,10 +375,14 @@ where
         let EthExecuteOutput { receipts, requests, gas_used } =
             self.execute_without_verification(block, total_difficulty)?;
 
-        // NOTE: we need to merge keep the reverts for the bundle retention
-        self.state.merge_transitions(BundleRetention::Reverts);
-
-        Ok(BlockExecutionOutput { state: self.state.take_bundle(), receipts, requests, gas_used })
+        let cache = core::mem::take(&mut self.state.cache);
```

Comment:
replace with self.state.take_cache() when https://github.com/bluealloy/revm/pull/1756 is merged